### PR TITLE
Add show player coordinates option

### PR DESCRIPTION
**Pull request by ryantam626**
For debugging purposes.

Also ignore some files generated by my IDE.

### CODE REVIEW

**Comment by CptWesley** [comment]
Haven't tried this myself yet, but does this play nicely with the different minimap shapes?

**Comment by ryantam626** [comment]
Nope, it most certainly does not. It's currently sits at the top right corner.

Do you want me to move it to the center, just below the player's marker (when map is not panned)

**Comment by CptWesley** [comment]
I think that might be better. That way it should work for all kinds of weird minimap shapes.

**Comment by ryantam626** [comment]
Gotcha - updated.

Panned
image


Overlay, diamond
image

File: overwolf/src/Minimap/drawPlayerCoordinates.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        const mapPos = toMinimapCoordinate(mapCenterPosition, playerPosition, ctx.canvas.width, ctx.canvas.height, zoomLevel, 1);
+        const imgPosCorrected = {
+            x: mapPos.x / zoomLevel - offset.x / zoomLevel + center.x,
+            y: mapPos.y / zoomLevel - offset.y / zoomLevel + center.y,
+        };
+
+        const textX = imgPosCorrected.x - 50;
+        const textY = imgPosCorrected.y + 30;
+        console.log(textX, textY);
```
Comment (pvdstel) [inline overwolf/src/Minimap/drawPlayerCoordinates.ts:27]: This log should be removed.

**Review by pvdstel — CHANGES_REQUESTED** [review]
Since the setting you added (`showPlayerCoordinates`) affects rendering, it should be added to the [`updateOn` array](https://github.com/ryantam626/NewWorldMinimap/blob/ebbdaefcaed71b1d1f0dc89bceff0e7443016a6c/overwolf/src/Minimap/useMinimapRenderer.ts#L270) (preferably with the alphabetic sorting). Adding the value `appContext.settings.showPlayerCoordinates` will automatically trigger a re-render, showing (or hiding) the coordinates.

---

If combined with navigation, the text starts looking quite funky. Haven't found the exact cause yet. I think it's `strokeStyle` from `drawMapNavigation`, which may have to be reset.

image

---

It doesn't play nice with `iconScale`. I'm not sure if this is too relevant, since I essentially consider this a debug setting. What do you think @CptWesley ?

image

---

The rendering changes depending on if `showText` is enabled. When it's on, it starts looking like this:

image

Haven't figured out the reason for that happening yet. It's certainly strange. It does make the text look exactly like the other text, so I imagine the styling/positioning there plays a role.

---

Some discussion about the location of the implementation: I think the current implementation (drawing the coordinates in a separate file) is fine. Some alternatives to consider, maybe for the future:

- Adding the coordinate rendering to `drawMapPlayer`.
- Moving the condition for rendering to `useMinimapRenderer/draw`. That would skip the function call and the deconstruction of arguments, and would make the added field to `MapRenderParameters` unnecessary.

Nothing you need to change here, just me thinking out loud 😄

---

A final point to think about, also not really an actionable point: this setting is a global setting, so it technically doesn't belong in the _This window_ tab in the app settings. I'd prefer to add another tab, _Advanced_ (or something) where this checkbox can live, especially if we consider this a debug setting. That tab could then also have the magical _Reset all settings_ button. But that's just to think about for the future.

**Comment by ryantam626** [comment]
Good spots @pvdstel - thanks!

Screeshot after latest updates.
Pan + nav on.
image

Show text on.
image

Non-default icon scale.
image

Non-default icon scale + show text on.
image

In-game truncated map style.
image

**Comment by CptWesley** [comment]
Your latest screenshots seem to somehow override the strokestyle of the icon text. Perhaps we should just extract the style-setting pieces of code to seperate functions so we can just do something like `setTextStyle()` and `setLineStyle()` or something similar to avoid these kind of issues in the future.

**Pull request merged**